docs/library/IPs/template_ip/index.rst

```diff
@@ -0,0 +1,119 @@
+:orphan:
+
+.. _template_ip:
+
+IP Template
+================================================================================
+
+.. hdl-component-diagram::
```

Comment:
This is not compatible with the current testbenches repo structure.
Valid for all `hdl-*` directives
I would remove from any template.

After we rework the repo structure, we can consider implementing the custom directives here also.

Reply:
Should I create some testbenches-* directives?

Also, is there a way to point to diagrams from the HDL repo instead of having a copy of them in Testbenches?

Reply:
I believe it's fair to reuse the same directive,
The `hdl-component-diagram` `hdl-interfaces` `hdl-parameters` are compatible with this repo, just need to update the `option` path to point a valid (V)IP.

I suggest updating the template hdl directives `path` options from `library/spi_engine/spi_engine_execution` and `library/axi_ad9783`  to SPI VIP `common/sv/spi_vip`  in the `library/xilinx/clk_vip/index.rst` file lines 9, 40, 51. (and after the repo is restructured, update again to `library/spi_vip`).

I tested it here, after the edit, I built the SPI VIP `(cd common/sv/spi_vip ; make)` to generate ` common/sv/spi_vip/component.xml` and then built the doc.
![image](https://github.com/user-attachments/assets/b3eef69d-f55a-4e61-a1a3-87e2e5fffa91).

About the `hdl-regmap` directive, I don't think the VIP will have a regmap, instead they are/shoud_be managed by an API in SystemVerilog (method calls). But that is something to discuss, and if we allow VIPs regmaps, we could use the same structure as the hdl repo: store it in `docs/regmap/adi_<regmap>.txt`

However, why was `template_ip/index.rst` renamed to `clk_vip/index.rst`?

> Also, is there a way to point to diagrams from the HDL repo instead of having a copy of them in Testbenches?

Image diagrams? like 
```
.. image:: block_diagram.svg
   :alt: AXI AD7616 block diagram
```
Relative paths are valid, like ```.. image:: ../../../../docs/library/axi_ad7616/block_diagram.svg```, just not sure if we should.